lib/src/value/calculation.dart

```diff
@@ -158,6 +382,197 @@ class SassCalculation extends Value {
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("rem", args);
+    }
+    _verifyCompatibleNumbers(args);
+
+    var result = dividend.modulo(modulus);
+    if ((modulus.value < 0 && dividend.value >= 0) ||
+        (modulus.value >= 0 && dividend.value < 0)) {
+      if (modulus.value.isInfinite) return dividend;
+      if (result is SassNumber && result.value == 0) return SassNumber(-0.0);
+      return result.minus(dividend);
+    }
+    return result;
+  }
+
+  /// Creates a `mod()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    strategyOrNumber = _simplify(strategyOrNumber);
+    numberOrStep = numberOrStep.andThen(_simplify);
+    step = step.andThen(_simplify);
+
+    var args = (strategyOrNumber, numberOrStep, step);
+
+    switch (args) {
+      case (SassNumber number, null, null):
+        return SassNumber(number.value.round().toDouble())
+            .coerceToMatch(number);
+
+      case (SassString strategy, SassNumber(), null)
+          when ({'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text)):
```

Comment:
```suggestion
      case (SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'), SassNumber(), null):
```

Reply:
I searched in the [pattern file](https://dart.dev/language/pattern-types) and couldn't find an example like this, this is genius 🥇